core-java-modules/core-java-io/src/test/java/com/baeldung/extension/ExtensionFromMimeTypeUnitTest.java

```diff
@@ -0,0 +1,47 @@
+package com.baeldung.extension;
+
+import static org.junit.Assert.assertEquals;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.tika.mime.MimeType;
+import org.apache.tika.mime.MimeTypeException;
+import org.apache.tika.mime.MimeTypes;
+import org.junit.Test;
+
+import com.j256.simplemagic.ContentInfo;
+
+public class ExtensionFromMimeTypeUnitTest {
+    public static final String PNG_EXT = "image/png";
+    @Test
+    public void whenUsingTika_thenGetFileExtension() throws MimeTypeException {
+        MimeTypes allTypes = MimeTypes.getDefaultMimeTypes();
+        MimeType type = allTypes.forName(PNG_EXT);
+        String extension = type.getExtension();
+        assertEquals(".png", extension);
+    }
+
+    @Test
+    public void  whenUsingJodd_thenGetFileExtension() {
+        String fileExtension = jodd.net.MimeTypes.findExtensionsByMimeTypes(PNG_EXT,false)[0];
+        assertEquals("png", fileExtension);
+    }
+
+    @Test
+    public void whenUsingMimetypesFileTypeMap_thenGetFileExtension() {
+        ContentInfo contentInfo = new ContentInfo("", PNG_EXT, "", true);
+        String[] fileExtensions = contentInfo.getFileExtensions();
+        assertEquals("png", fileExtensions[0]);
+    }
+
+    @Test
+    public void whenUsingCustomLogic_thenGetFileExtension() {
+        Map<String, String> mimeToExtensionMap = new HashMap<>();
```

Comment:
There is a Many-Many relationship between mime types and file extensions, so the map should really be `Map<String, List<String>>` for example, audio/midi can have extension .mid or .midi

Reply:
Done